sdk/dotnet/Pulumi.Automation/Events/EventLogWatcher.cs

```diff
@@ -8,20 +8,19 @@
 
 namespace Pulumi.Automation.Events
 {
-    internal class EventLogWatcher : IAsyncDisposable
+    internal class EventLogWatcher : IDisposable
     {
         private readonly LocalSerializer _localSerializer = new LocalSerializer();
-        private readonly CancellationTokenSource _internalCancellationTokenSource = new CancellationTokenSource();
-        private readonly CancellationToken _externalCancellationToken;
         private readonly Action<EngineEvent> _onEvent;
-        private readonly Task _pollingTask;
-
         private const int _pollingIntervalMilliseconds = 100;
 
-        // We keep track of the length we have already read from the file
-        private long _previousLength = 0;
-
+        // We keep track of the last position in the file.
+        private long _position = 0;
         public string LogFile { get; }
+        private Task _pollingTask;
+        private CancellationTokenSource _internalCancellationTokenSource = new CancellationTokenSource();
+
+        private CancellationToken? _cancellationToken;
```

Comment:
This field doesn't seem to be used for other than assigning to it in `PollForEvents`, so it could be dropped